File: viewpump/src/main/java/io/github/inflationx/viewpump/FallbackViewCreationInterceptor.java
package io.github.inflationx.viewpump;

import android.view.View;

class FallbackViewCreationInterceptor implements Interceptor {

    @Override
    public InflateResult intercept(Chain chain) {
        InflateRequest request = chain.request();
        FallbackViewCreator viewCreator = request.fallbackViewCreator();
        View fallbackView = viewCreator.onCreateView(request.parent(), request.name(), request.context(), request.attrs());

        return InflateResult.builder()
                .view(fallbackView)
                .name(fallbackView != null ? fallbackView.getClass().getName() : request.name())
                .context(request.context())
                .attrs(request.attrs())
                .build();
    }
}

File: viewpump/src/main/java/io/github/inflationx/viewpump/-FallbackViewCreationInterceptor.kt
@file:JvmName("-FallbackViewCreationInterceptor")
package io.github.inflationx.viewpump

internal class FallbackViewCreationInterceptor : Interceptor {

  override fun intercept(chain: Interceptor.Chain): InflateResult {
    val request = chain.request()
    val viewCreator = request.fallbackViewCreator()
    val fallbackView = viewCreator.onCreateView(request.parent(), request.name(), request.context(),
        request.attrs())

    return InflateResult.builder()
        .view(fallbackView)
        .name(fallbackView?.javaClass?.name ?: request.name())
        .context(request.context())
        .attrs(request.attrs())
        .build()
  }
}
